tortoise/tortoise.go

```diff
@@ -498,12 +483,15 @@ func (t *turtle) processBallot(ctx context.Context, ballot *types.Ballot) error
 			continue
 		}
 		if _, exist := opinion[blk]; !exist {
-			nvote := simplifyVote(vote).Multiply(voteWeight)
-			opinion[blk] = nvote
+			opinion[blk] = vote.copy()
 		}
 	}
+
+	logger.With().Debug("adding or updating ballot opinion", log.Stringer("ballot_weight", weight))
+
+	t.BallotWeight[ballot.ID()] = weight
+	t.BlockLayer[types.BlockID(ballot.ID())] = ballot.LayerIndex
```

Comment:
why is this needed? we should only look up t.BlockLayer when we look to trim blocks in the opinion map. 
we should only do this when processing blocks (not ballots)

Reply:
it is needed for unit tests that are not using db. it was previously done in handleLayer that reads blocks from database, if later there will be separate code path for blocks there should be processBlocks then

Reply:
i added processBlocks function

Reply:
it's handled in `handleLayer` because that's the only place we will potentially see Block. this is the only time we know about what blocks belong to which layer. it still doesn't make sense to do it here. BallotID will be totally different from BlockID/ProposalID in a few more PRs. 

can you change the unit test instead?